File: src/main/java/ar/edu/unq/epers/unidad4/exception/MuchoPesoException.java
package ar.edu.unq.epers.unidad4.exception;

import ar.edu.unq.epers.unidad4.model.Item;
import ar.edu.unq.epers.unidad4.model.Personaje;

final public class MuchoPesoException extends RuntimeException {

    private static final long serialVersionUID = 1L;
    private final Personaje personaje;
    private final Item item;

    public MuchoPesoException(Personaje personaje, Item item) {
        this.personaje = personaje;
        this.item = item;
    }

    @Override
    public String getMessage() {
        return "El personaje [" + personaje.getNombre() + "] no puede recoger [" + item.getNombre() + "] porque cargar mucho peso ya";
    }

}

File: src/main/kotlin/ar/edu/unq/epers/unidad4/exception/MuchoPesoException.kt
package ar.edu.unq.epers.unidad4.exception

import ar.edu.unq.epers.unidad4.model.Item
import ar.edu.unq.epers.unidad4.model.Personaje

class MuchoPesoException(private val personaje: Personaje, private val item: Item) : RuntimeException() {

    override val message: String?
        get() = "El personaje [$personaje] no puede recoger [$item] porque cagar mucho peso ya"

    companion object {

        private val serialVersionUID = 1L
    }

}